src/state/nftMarket/hooks.ts

```diff
@@ -8,28 +8,40 @@ import erc721Abi from 'config/abi/erc721.json'
 import { useSWRMulticall } from 'hooks/useSWRContract'
 import { EMPTY_ARRAY, EMPTY_OBJECT } from 'utils/constantObjects'
 import { getPancakeProfileAddress } from 'utils/addressHelpers'
+import useSWRImmutable from 'swr/immutable'
+import { useSWRConfig } from 'swr'
 
-import { fetchCollection, fetchCollections, fetchNewPBAndUpdateExisting } from './reducer'
+import { fetchNewPBAndUpdateExisting } from './reducer'
 import { State } from '../types'
-import { NftActivityFilter, NftFilter, NftToken } from './types'
+import { ApiCollections, NftActivityFilter, NftFilter, NftToken } from './types'
+import { getCollection, getCollections } from './helpers'
 
 const DEFAULT_NFT_ORDERING = { field: 'currentAskPrice', direction: 'asc' as 'asc' | 'desc' }
 const DEFAULT_NFT_ACTIVITY_FILTER = { typeFilters: [], collectionFilters: [] }
 
 export const useFetchCollections = () => {
   const dispatch = useAppDispatch()
+  const { mutate } = useSWRConfig()
   useEffect(() => {
-    dispatch(fetchCollections())
-  }, [dispatch])
+    const fetchCollections = async () => {
+      const collections = await getCollections()
+      mutate(['nftMarket', 'collections'], collections)
+    }
+    fetchCollections()
+  }, [mutate, dispatch])
 }
 
 export const useFetchCollection = (collectionAddress: string) => {
   const dispatch = useAppDispatch()
+  const { mutate } = useSWRConfig()
   useEffect(() => {
     if (collectionAddress) {
-      dispatch(fetchCollection(collectionAddress))
+      mutate(['nftMarket', 'collections'], async (collections) => {
+        const collectionData = await getCollection(collectionAddress)
+        return { ...collections, ...collectionData }
+      })
     }
-  }, [dispatch, collectionAddress])
+  }, [mutate, dispatch, collectionAddress])
```

Comment:
simply refactor use swr and combine them with `useGetCollections` or `useGetCollection`

Reply:
Done